elephant/conversion.py

```diff
@@ -152,3 +155,679 @@ def binarize(spiketrain, sampling_rate=None, t_start=None, t_stop=None,
+    transform the binned spike train to a boolean matrix or a matrix with
+    counted time points.
+
+    A binned spike train represents the occurrence of spikes in a certain time
+    frame.
+    I.e., a time series like [0.5, 0.7, 1.2, 3.1, 4.3, 5.5, 6.7] is
+    represented as [0, 0, 1, 3, 4, 5, 6]. The outcome is dependent on given
+    parameter such as size of bins, number of bins, start and stop points.
+
+    A boolean matrix represents the binned spike train in a binary (True/False)
+    manner.
+    It's rows represent the number of spike trains
+    and the columns represent the binned index position of a spike in a
+    spike train.
+    The calculated matrix columns contains **Trues**, which indicate
+    a spike.
+
+    A matrix with counted time points is calculated the same way, but its
+    columns contain the number of spikes that occurred in the spike train(s).
+    It counts the occurrence of the timing of a spike in its respective
+    spike train.
+
+    Parameters
+    ----------
+    spiketrains : List of `neo.SpikeTrain` or a `neo.SpikeTrain` object
+        Object or list of `neo.core.SpikeTrain` objects to be binned.
+    binsize : quantities.Quantity
+        Width of each time bin.
+        Default is `None`
+    num_bins : int
+        Number of bins of the binned spike train.
+        Default is `None`
+    t_start : quantities.Quantity
+        Time of the first bin (left extreme; included).
+        Default is `None`
+    t_stop : quantities.Quantity
+        Stopping time of the last bin (right extreme; excluded).
+        Default is `None`
+    store_matrix : bool
+        If set to **True** matrix with counted time points will be stored in
+         memory.
+        If set to **False** matrix will always be calculated on demand.
+        This boolean indicates that the matrix with counted time points
+        will be stored in memory.
+        Default is False.
+
+    See also
+    --------
+    _convert_to_binned
+    spike_indices
+    to_bool_array
+    to_array
+
+    Notes
+    -----
+    There are four cases the given parameters must fulfill.
+    Each parameter must be a combination of following order or it will raise
+    a value error:
+    * t_start, num_bins, binsize
+    * t_start, num_bins, t_stop
+    * t_start, bin_size, t_stop
+    * t_stop, num_bins, binsize
+
+    It is possible to give the SpikeTrain objects and one parameter
+    (:attr:`num_bins` or :attr:`binsize`). The start and stop time will be
+    calculated from given SpikeTrain objects (max start and min stop point).
+    Missing parameter will also be calculated automatically.
+    All parameters will be checked for consistency. A corresponding error will
+    be risen, if one of the four parameters does not match the consistency
+    requirements.
+
+    """
+
+    def __init__(self, spiketrains, binsize=None, num_bins=None, t_start=None,
+                 t_stop=None, store_matrix=False):
```

Comment:
`store_matrix` should be `store_array`, for consistency with the renamed methods.